manipulation/src/main/java/org/jboss/gm/manipulation/actions/PublishingRepositoryAction.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ * <p>
+ * Licensed under the Eclipse Public License version 1.0, available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.jboss.gm.manipulation.actions;
+
+import org.gradle.api.Action;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.repositories.MavenArtifactRepository;
+import org.gradle.api.credentials.HttpHeaderCredentials;
+import org.gradle.api.publish.PublishingExtension;
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin;
+import org.gradle.authentication.http.HttpHeaderAuthentication;
+
+/**
+ * Adds a publishing repository specific to PNC environment.
+ * <p>
+ * System properties "AProxDeployUrl" and "accessToken" has to be defined during build.
+ * <p>
+ * Is equivalent to following gradle snippet:
+ *
+ * <pre>
+ * publishing {
+ *         repositories {
+ *             maven {
+ *                 name = "PNC"
+ *                 url = System.getProperty('AProxDeployUrl')
+ *                 credentials(HttpHeaderCredentials) {
+ *                     name = "Authorization"
+ *                     value = "Bearer " + System.getProperty('accessToken')
+ *                 }
+ *                 authentication {
+ *                     header(HttpHeaderAuthentication)
+ *                 }
+ *             }
+ *         }
+ *     }
+ * </pre>
+ */
+public class PublishingRepositoryAction implements Action<Project> {
+
+    public static final String URL_SYSTEM_PROPERTY = "AProxDeployUrl";
+
+    public static final String ACCESS_TOKEN_SYSTEM_PROPERTY = "accessToken";
+
+    @Override
+    public void execute(Project project) {
+        String pncDeployUrl = System.getProperty(URL_SYSTEM_PROPERTY);
```

Comment:
Are the system properties populated with environment variables? - I believe AProxDeployUrl is an env variable in the pod.

Reply:
I am starting to think that the plugin needs to check whether `maven` or `maven-publish` plugin is configured and act accordingly. If none is configured (which I am assuming is very unlikely), we can just choose one.

Reply:
Could we just remove any existing instance / configuration and replace with our known configuration?

Reply:
> Are the system properties populated with environment variables? - I believe AProxDeployUrl is an env variable in the pod.

I don't think so, I set it via -D switch in the gradle command. But it would be handy improvement - first looks for system prop and then fallback to `System.getenv()`?

Reply:
> I am starting to think that the plugin needs to check whether `maven` or `maven-publish` plugin is configured and act accordingly. If none is configured (which I am assuming is very unlikely), we can just choose one.

+1

Reply:
> Could we just remove any existing instance / configuration and replace with our known configuration?

I am not sure, we'll need to check this.

Reply:
> Are the system properties populated with environment variables? - I believe AProxDeployUrl is an env variable in the pod.

We should probably do something similar to what is being done in the other plugin where Apache Commons Config is used in order to read config from both system properties and environment variables.